pkg/deployer/rte/rte.go

```diff
@@ -142,22 +142,20 @@ func Remove(logger *log.Logger, opts Options) error {
 	mf = mf.EnforceNamespace()
 	logger.Printf("manifests loaded")
 
-	// no need to update in the remove path
-	if err := hp.DeleteObject(mf.DaemonSet); err != nil {
+	// since we created everything in the namespace, we can just do
```

Comment:
Since the remove logic is heavily relying on the fact that all objects are under the same ns, don't you think it is better to invoke ``` mf.EnforceNamespace()``` inside ```GetManifests()```? this way you keep the user (and by user I mean other contributors) from dropping the call to ```mf.EnforceNamespace()``` and avoid potential issues.

Reply:
The (admittedly undocumented) intent of `GetManifests` is to provide the pristine versions of the manifests. The trasformations should be explicit as possible, so we should add another helper (and write the docstrings to make the intent expliit) rather then coalesce more responsabilities into `GetManifests`.

I agree about reducing the margin for errors, once we have a more solid design we should indeed add higher level helpers.
In general, catching these errors calls for more tests (unit, e2e) more than else.

So I think we should add automated tests first, let the design solidify a bit meantime, and then revisit the APIs and/or fix the existing functions.